lib/user/user-client.js

```diff
@@ -29,13 +31,65 @@ var xhr = function(_options) {
   return deferred.promise;
 };
 
-var storeProfile = function(profileData) {
-  localStorage.setItem('fh.wfm.profileData', JSON.stringify(profileData));
+/**
+* Encrypt the users profile data using AES with a hash of the session key and store the result in localStorage.
+* @param profileData {object} - the plaintext to encrypt.
+* @param sessionToken {string} - the secret key to encrypt with.
+*/
+var storeProfile = function(profileData, sessionToken) {
+  if (sessionToken && profileData) {
+
+    var hashedSessionKey = sha256.hash(sessionToken);
+    // handle failed hashing of session token
+    if (hashedSessionKey instanceof Error) {
+      storeLocalStorage('fh.wfm.profileData', null, hashedSessionKey.toString());
+      return;
+    }
+
```

Comment:
better create some function
```js
function useLocalStorage(obj)
  if (obj instanceof Error) {
    storeLocalStorage('fh.wfm.profileData', null, obj.toString());
    return;
 }
```
And then instead of 
```js
if(true){return}
...
if(true){return}
```
use `if(){}else if() else{}` statement